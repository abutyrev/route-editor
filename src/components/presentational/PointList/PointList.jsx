import React from "react";
import { DragDropContext, Droppable, Draggable } from "react-beautiful-dnd";

const PointList = ({ points, deletePoint, onDragEnd }) => {
  const list = points.map((point, index) => {
    return (
      <Draggable key={point.id} draggableId={point.id} index={index}>
        {provided => (
          <div
            className="collection-item"
            ref={provided.innerRef}
            {...provided.draggableProps}
            {...provided.dragHandleProps}
          >
            <span>
              {`${index + 1}. ${point.point.name}`}
              <a
                href="#!"
                className="secondary-content"
                onClick={deletePoint.bind(null, point.id)}
              >
                <i className="material-icons">delete</i>
              </a>
            </span>
          </div>
        )}
      </Draggable>
    );
  });

  return (
    <DragDropContext onDragEnd={onDragEnd}>
      <Droppable droppableId="droppable">
        {provided => (
          <div ref={provided.innerRef} className="point-list collection">
            {list}
          </div>
        )}
      </Droppable>
    </DragDropContext>
  );
};

export default PointList;
