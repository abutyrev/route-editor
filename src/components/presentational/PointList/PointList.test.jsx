import React from "react";
import { shallow, mount } from "enzyme";

import PointList from "./PointList";

describe("PointList component tests", () => {
  const deletePoint = jest.fn(),
    onDragEnd = jest.fn();

  it("should render correctly", () => {
    const props = {
        points: [],
        deletePoint,
        onDragEnd
      },
      PointListComponent = shallow(<PointList {...props} />);
    expect(PointListComponent).toMatchSnapshot();
  });

  it("should render correct list item", () => {
    const testProps = {
        points: [
          {
            id: 1,
            point: null
          }
        ],
        deletePoint,
        onDragEnd
      },
      PointListComponent = mount(<PointList {...testProps} />);
    expect(
      PointListComponent.find(".collection-item")
        .find("span")
        .text()
    ).toEqual("Точка маршрута 1delete");
  });

  it("check deletePoint callback", () => {
    const testProps = {
        points: [
          {
            id: 1,
            point: null
          }
        ],
        deletePoint,
        onDragEnd
      },
      PointListComponent = mount(<PointList {...testProps} />),
      a = PointListComponent.find(".collection-item").find("a");
    a.simulate("click");
    expect(deletePoint).toHaveBeenCalledTimes(1);
  });

/*   it("check onDragEnd callback", () => {
    const testProps = {
        points: [
          {
            id: 1,
            point: null
          }
        ],
        deletePoint,
        onDragEnd
      },
      PointListComponent = mount(<PointList {...testProps} />),
      item = PointListComponent.find('.collection-item');
      item.simulate('dragend');
    expect(onDragEnd).toHaveBeenCalledTimes(1);
  }); */
});
