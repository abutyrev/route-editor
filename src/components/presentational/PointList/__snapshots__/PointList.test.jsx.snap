// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PointList component tests should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <PointList
    deletePoint={[MockFunction]}
    onDragEnd={[MockFunction]}
    points={Array []}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <Connect(Droppable)
        direction="vertical"
        droppableId="droppable"
        ignoreContainerClipping={false}
        isCombineEnabled={false}
        isDropDisabled={false}
        type="DEFAULT"
      >
        [Function]
      </Connect(Droppable)>,
      "onDragEnd": [MockFunction],
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": [Function],
        "direction": "vertical",
        "droppableId": "droppable",
        "ignoreContainerClipping": false,
        "isCombineEnabled": false,
        "isDropDisabled": false,
        "type": "DEFAULT",
      },
      "ref": null,
      "rendered": [Function],
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <Connect(Droppable)
          direction="vertical"
          droppableId="droppable"
          ignoreContainerClipping={false}
          isCombineEnabled={false}
          isDropDisabled={false}
          type="DEFAULT"
        >
          [Function]
        </Connect(Droppable)>,
        "onDragEnd": [MockFunction],
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": [Function],
          "direction": "vertical",
          "droppableId": "droppable",
          "ignoreContainerClipping": false,
          "isCombineEnabled": false,
          "isDropDisabled": false,
          "type": "DEFAULT",
        },
        "ref": null,
        "rendered": [Function],
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
